File: project2/scripts/bucklingbeam1.cpp
/* script solving project 2 b) Finding the similarity transformations needed
to diagonalize the buckling beam matrix */

#include <iostream>
#include <cstdlib>
#include <fstream>
#include <iomanip>
#include <armadillo>
#include <cmath>
#include <vector>
#include "time.h"
#include "../include/utils.hpp"

int main(){

  int eps = -8;
  int N_min = 5;
  int N_step = 10;
  int N_points = 13;
  std::string filename = "dat/SimTransCount.csv";

  SimTransCount(eps, N_min, N_step, N_points, filename);

  return 0;
}

File: project2/scripts/bucklingbeam1.py
# reading file from c++

import numpy as np
import matplotlib.pyplot as plt

X = np.genfromtxt("dat/SimTransCount.csv", delimiter=',',skip_header=1)
#x = np.linspace(0, 1, X.shape[0]) # generating the x-arrays

plt.figure()

plt.subplot(211)
plt.plot(X[1:,0], X[1:,1], 'g', X[1:,0], X[1:,0]**2, 'r--')
plt.xscale("log")
plt.yscale("log")
plt.title("Buckling beam, similarity transformations")
plt.xlabel("N (dimensionality of matrix)")
plt.ylabel("Similarity transformations")

plt.subplot(212)
plt.plot(X[1:,0], X[1:,2], 'b')
plt.title("CPU time as function of dimensionality")
plt.xlabel("N (dimensionality of matrix)")
plt.ylabel("CPU time")

plt.show()
